### PR TITLE
Reivew ch7, ch8 and ch10 text and code

### LINKED ISSUE #231: Ch7 - Clarify applicative validation
# Issue 1

`nonEmpty` is defined and used in `validatePerson` like so:

```purs
nonEmpty :: String -> Either String Unit
nonEmpty "" = Left "Field cannot be empty"
nonEmpty _  = Right unit

validatePerson :: Person -> Either String Person
validatePerson p =
  person <$> (nonEmpty p.firstName *> pure p.firstName)
         <*> (nonEmpty p.lastName  *> pure p.lastName)
         <*> pure p.address
         <*> pure p.phones
```

One wonders why `nonEmpty` isn't defined instead as:

```purs
nonEmpty :: String -> Either String String
nonEmpty "" = Left "Field cannot be empty"
nonEmpty s  = Right s

validatePerson :: Person -> Either String Person
validatePerson p =
  person <$> nonEmpty p.firstName
         <*> nonEmpty p.lastName
         <*> pure p.address
         <*> pure p.phones
```

# Issue 2

Here we're also introduced for the first time to `(*>)` operator which in my opinion is explained very poorly. Since `(<*>)` was already explained in the previous section as an alias for `apply` I think `(*>)` is much better presented as in the docs, where we're told that (*>) keeps the results of the second computation, and throws away the results of the first. Maybe the signature should be given too.

# Issue 3

The code uses the field `homeAddress` instead of `address`.

### PR DESCRIPTION
**Pull request by razcore-art**
This is a draft as I only reviewed ch7.

Closes #231

### CODE REVIEW

**Comment by milesfrain** [comment]
Looks great.
I'll need to fix the CI issue reported in #246

**Comment by razcore-art** [comment]
@milesfrain I did a bit more than just update the code as I saw certain areas in the text that could be improved too. I also have this markdown vim plugin that enforces some whitespace rules and I used it to reformat the text a bit, but that shouldn't be a problem, it just adds some newlines to separate lists & code from paragraphs.

I also updated `.gitignore` to include `/book`, `/index.html` and a couple of other things that could be committed by mistake while working on the code with `spago`, `mdbook` and `parcel`. You can pick and choose the commits to merge, right? :) 

I replaced `unV` with `toEither` and improved hopefully, the formatting here and there. Anyway, I await your review.

File: exercises/chapter10/src/Data/AddressBook.purs [hunk 1]
```diff
@@ -21,19 +25,12 @@ data PhoneType
...
-  encodeJson = genericEncodeJson
-
-instance decodeJsonPhoneType :: DecodeJson PhoneType where
-  decodeJson = genericDecodeJson
+-- ANCHOR: PhoneType_generic
+derive instance genericPhoneType    :: Generic    PhoneType _
+instance        encodeJsonPhoneType :: EncodeJson PhoneType where encodeJson = genericEncodeJson
+instance        decodeJsonPhoneType :: DecodeJson PhoneType where decodeJson = genericDecodeJson
+-- ANCHOR_END: PhoneType_generic
+instance        showPhoneType       :: Show       PhoneType where show       = genericShow
```
Comment (milesfrain) [inline exercises/chapter10/src/Data/AddressBook.purs:33]: Wondering if `genericShow` should be used for `showPhoneType` in Ch7 and Ch8 too. Will investigate further after merging this PR and #240.

File: text/chapter10.md [hunk 2]
```diff
@@ -54,17 +54,13 @@ This function has the correct runtime representation for the function type `Stri
 We can assign this type to the function with the following foreign import declaration:
 
 ```haskell
-module Test.URI where
-
-foreign import encodeURIComponent :: String -> String
+{{#include ../exercises/chapter10/test/URI.purs}}
```
Comment (milesfrain) [inline text/chapter10.md:57]: Nice! Even better than setting-up an anchor.

File: text/chapter10.md [hunk 3]
```diff
@@ -1009,14 +1036,15 @@ mkAddressBookApp =
...
 processItem :: Json -> Either String Person
 processItem item = do
   jsonString <- lmap ("No string in local storage: " <> _) $ decodeJson item
-  j <- lmap ("Cannot parse JSON string: " <> _) $ jsonParser jsonString
-  (p :: Person) <- lmap ("Cannot decode Person: " <> _) $ decodeJson j
-  pure p
+  j          <- lmap ("Cannot parse JSON string: "   <> _) $ jsonParser jsonString
+  lmap               ("Cannot decode Person: "       <> _) $ decodeJson j
 ```
+
```
Comment (milesfrain) [inline text/chapter10.md:1047]: Code anchors would be good for these sections too, but that can be tackled in another PR.

**Comment by milesfrain** [comment]
This looks really good. Excellent attention to detail.

File: text/chapter10.md (hunk 3 above)
Comment (razcore-art) [reply text/chapter10.md:1047]: Yeah, I don't remember why I skipped this one.

**Pull request merged**